[700] Update license files for bundled jars

vinishjail97 opened the pull request:
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

Addressing feedback from @zabetak and @jbonofre from rc2 vote thread. 
https://github.com/apache/incubator-xtable/issues/700 

## Brief change log

*(for example:)*
- *Update license files for bundled jars*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is a trivial rework / code cleanup without any test coverage.

zabetak reviewed (changes requested):
Hey @vinishjail97 , thanks for taking up this task. 

It seems that there is some misunderstanding around the license content. The `licenses` folder is not **always** redundant. The fat/bundled jars must include it but not the rest.

Moreover, in all cases there should be only one license entry point (i.e., `LICENSE`). The jars must not have 2 or more (e.g., both `LICENSE` and `LICENSE-bundled`).

If something is not clear let me know and I will try to clarify more.

vinishjail97 commented:
> Hey @vinishjail97 , thanks for taking up this task.
> 
> It seems that there is some misunderstanding around the license content. The `licenses` folder is not **always** redundant. The fat/bundled jars must include it but not the rest.
> 
> Moreover, in all cases there should be only one license entry point (i.e., `LICENSE`). The jars must not have 2 or more (e.g., both `LICENSE` and `LICENSE-bundled`).
> 
> If something is not clear let me know and I will try to clarify more.

I have updated the license files and reverted the other changes. 
For bundled jars I'm replacing the license file content using this configuration. 
https://github.com/apache/incubator-xtable/blob/main/xtable-aws/pom.xml#L121

jbonofre reviewed (commented):
At first glance, it looks good. Give me 10mn to build and check on the "final" artifact.
What about the NOTICE (for ALv2 bundled dependencies) ?

vinishjail97 commented:
> What about the NOTICE (for ALv2 bundled dependencies) ?

@jbonofre 
There's some confusion between NOTICE files being bloated and the guidance given in https://issues.apache.org/jira/browse/LEGAL-684. What changes do we need for NOTICE according to you?

jbonofre commented:
@vinishjail97 you are adding non ASF project in the `LICENSE` (like Jackson). If I agree that it's not needed to add `NOTICE` content for ASF dependencies, `NOTICE` content for ALv2 dependencies should be checked and included in the `NOTICE`.

Assuming that the bundled dependency itself contains no bundled sub-components under other licenses, so the ALv2 applies uniformly to all files, there is no need to modify LICENSE. However, for completeness it is useful to list the products and their versions, as is done for products under other licenses.
If the dependency supplies a NOTICE file, its contents must be analyzed and the relevant portions bubbled up into the top-level NOTICE file.

So, if you checked the content and you are good with that, it's fine for me.

vinishjail97 commented:
> @vinishjail97 you are adding non ASF projects in the `LICENSE` (like Jackson). If I agree that it's not needed to add `NOTICE` content for ASF dependencies, `NOTICE` content for ALv2 dependencies (third parties) should be checked and included in the `NOTICE`.
> 
> Assuming that the bundled dependency itself contains no bundled sub-components under other licenses, so the ALv2 applies uniformly to all files, there is no need to modify LICENSE. However, for completeness it is useful to list the products and their versions, as is done for products under other licenses. If the dependency supplies a NOTICE file, its contents must be analyzed and the relevant portions bubbled up into the top-level NOTICE file.
> 
> So, if you checked the content and you are good with that, it's fine for me.

Thanks for the comment JB it's more clear now. If I add a file similar to this one I found in iceberg aws bundle for the non ASG ALv2 dependencies, would it meet the requirements?    
https://github.com/apache/iceberg/blob/main/aws-bundle/NOTICE

vinishjail97 commented:
> @vinishjail97 you are adding non ASF projects in the `LICENSE` (like Jackson). If I agree that it's not needed to add `NOTICE` content for ASF dependencies, `NOTICE` content for ALv2 dependencies (third parties) should be checked and included in the `NOTICE`.
> 
> Assuming that the bundled dependency itself contains no bundled sub-components under other licenses, so the ALv2 applies uniformly to all files, there is no need to modify LICENSE. However, for completeness it is useful to list the products and their versions, as is done for products under other licenses. If the dependency supplies a NOTICE file, its contents must be analyzed and the relevant portions bubbled up into the top-level NOTICE file.
> 
> So, if you checked the content and you are good with that, it's fine for me.

@zabetak @jcamachor Can you confirm/ack this as well? I would like to get the dev vote thread aligned and pass easily for rc3.

zabetak commented:
Extract from the AL2 license:

_If the Work includes a "NOTICE" text file as part of its distribution, then any Derivative Works that You distribute must include a readable copy of the **attribution notices** contained within such NOTICE file, excluding those notices that do not pertain to any part of the Derivative Works_

In a nutshell, we much check the NOTICE file in our dependencies and if present, extract/copy the Copyright related information in the XTable NOTICE file. This is for projects under AL2 that are **not ASF projects**.

For ASF dependencies we don't need to do this as explained in [LEGAL-684](https://issues.apache.org/jira/browse/LEGAL-684).

The https://github.com/apache/iceberg/blob/main/aws-bundle/NOTICE is bloated with lots of useless information so I wouldn't say it is the best example.

vinishjail97 commented:
> Extract from the AL2 license:
> 
> _If the Work includes a "NOTICE" text file as part of its distribution, then any Derivative Works that You distribute must include a readable copy of the **attribution notices** contained within such NOTICE file, excluding those notices that do not pertain to any part of the Derivative Works_
> 
> In a nutshell, we much check the NOTICE file in our dependencies and if present, extract/copy the Copyright related information in the XTable NOTICE file. This is for projects under AL2 that are **not ASF projects**.
> 
> For ASF dependencies we don't need to do this as explained in [LEGAL-684](https://issues.apache.org/jira/browse/LEGAL-684).
> 
> The https://github.com/apache/iceberg/blob/main/aws-bundle/NOTICE is bloated with lots of useless information so I wouldn't say it is the best example.

Thanks for replying @zabetak, any idea what would be a good example? 
Every apache project I have looked at has it's own way and it has been difficult for me to interpret what is needed or not needed.

jbonofre commented:
@zabetak I disagree about Iceberg: for shaded jar or distribution (not code copy), it makes sense to copy ALv2 dependencies `NOTICE`. It's not only the copyright, it's also shaded dependencies in the dependency.
For code copy, I agree, the `NOTICE` should be analyzed to see if parts should be included or not.

jbonofre commented:
If it helps, another resource: https://infra.apache.org/licensing-howto.html

jbonofre commented:
@zabetak @vinishjail97 I propose to take a look on these examples I did in Polaris:
1. source distribution (code copy): https://github.com/apache/polaris/blob/main/LICENSE and https://github.com/apache/polaris/blob/main/NOTICE
2. binary distribution (bundling dependencies artifacts): https://github.com/apache/polaris/blob/main/quarkus/server/distribution/LICENSE and https://github.com/apache/polaris/blob/main/quarkus/server/distribution/NOTICE

Let me know if you want me to do a new pass on xtable.

zabetak commented:
There are some parts that I don't feel very strongly about so instead of disagreeing I will rather step back at this point and let @jbonofre make the final call.

jbonofre commented:
This PR is about `LICENSE`, and it looks good to me. So I propose to merge this PR and check on `NOTICE`. If @vinishjail97 did a pass/analyze about `NOTICE`, I'm fine with that.

vinishjail97 commented:
> There are some parts that I don't feel very strongly about so instead of disagreeing I will rather step back at this point and let @jbonofre make the final call.

@zabetak Can you approve the change as well? It's blocking the merge.
